WordPress/Classes/ViewRelated/Reader/ReaderTagCardCellViewModel.swift

```diff
@@ -0,0 +1,100 @@
+
+class ReaderTagCardCellViewModel: NSObject {
+
+    private typealias DataSource = UICollectionViewDiffableDataSource<Int, NSManagedObjectID>
+    private typealias Snapshot = NSDiffableDataSourceSnapshot<Int, NSManagedObjectID>
+
+    private weak var parentViewController: UIViewController?
+    private let slug: String
+    private weak var collectionView: UICollectionView?
+    private let isLoggedIn: Bool
+    private let cellSize: () -> CGSize?
+
+    private lazy var dataSource: DataSource? = {
+        guard let collectionView else {
+           return nil
+        }
+        return DataSource(collectionView: collectionView) { [weak self] collectionView, indexPath, objectID in
+            guard let post = try? ContextManager.shared.mainContext.existingObject(with: objectID) as? ReaderPost,
+                  let cell = collectionView.dequeueReusableCell(withReuseIdentifier: ReaderTagCell.classNameWithoutNamespaces(), for: indexPath) as? ReaderTagCell else {
+                return UICollectionViewCell()
+            }
+            cell.configure(parent: self?.parentViewController,
+                           post: post,
+                           isLoggedIn: self?.isLoggedIn ?? AccountHelper.isLoggedIn)
+            return cell
+        }
+    }()
+
+    private lazy var resultsController: NSFetchedResultsController<ReaderPost> = {
+        let fetchRequest = NSFetchRequest<ReaderPost>(entityName: ReaderPost.classNameWithoutNamespaces())
+        fetchRequest.sortDescriptors = [NSSortDescriptor(key: "sortRank", ascending: false)]
+        fetchRequest.fetchLimit = Constants.displayPostLimit
+        let resultsController = NSFetchedResultsController<ReaderPost>(fetchRequest: fetchRequest,
+                                                           managedObjectContext: ContextManager.shared.mainContext,
+                                                           sectionNameKeyPath: nil,
+                                                           cacheName: nil)
+        resultsController.delegate = self
+        return resultsController
+    }()
+
+    init(parent: UIViewController?, tag: ReaderTagTopic, collectionView: UICollectionView?, isLoggedIn: Bool, cellSize: @escaping @autoclosure () -> CGSize?) {
+        self.parentViewController = parent
+        self.slug = tag.slug
+        self.collectionView = collectionView
+        self.isLoggedIn = isLoggedIn
+        self.cellSize = cellSize
+
+        super.init()
+
+        resultsController.fetchRequest.predicate = NSPredicate(format: "topic = %@ AND isSiteBlocked = NO", tag)
+        collectionView?.delegate = self
+    }
+
+    func fetchTagTopics() {
```

Comment:
Nit: I think this should probably be `fetchTagPosts` or `fetchTopicPosts`, right?

Reply:
Yep, that's probably a better name for it. I renamed it in https://github.com/wordpress-mobile/WordPress-iOS/pull/23057/commits/e030eebf7ad50fc4b310e7aa0930b00f2221768a.